### PR TITLE
Added TGA, BMP and PSD readers.  Added TGA and BMP writers.

### PR DESCRIPTION
**Pull request by ggarra13**
They are very light as they rely on two auxiliary .h files (stb_image.h and stb_image_writer.h).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -42,6 +42,8 @@ set(TLRENDER_QT5 FALSE CACHE BOOL "Enable support for Qt5")
 set(TLRENDER_PROGRAMS TRUE CACHE BOOL "Build programs")
 set(TLRENDER_EXAMPLES TRUE CACHE BOOL "Build examples")
 set(TLRENDER_TESTS TRUE CACHE BOOL "Build tests")
+set(TLRENDER_GCOV FALSE CACHE BOOL "Enable gcov code coverage")
```
Comment (darbyjohnston) [inline CMakeLists.txt:45]: Could you check that you are synced with that latest? These coverage changes don't seem to be related to the PR.

File: lib/tlIO/STB.h [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                const std::string& fileName,
+                const file::MemoryRead*) override;
+            io::VideoData _readVideo(
+                const std::string& fileName,
+                const file::MemoryRead*,
+                const otime::RationalTime&,
+                uint16_t layer) override;
+        };
+
+        //! SGI writer.
```
Comment (darbyjohnston) [inline lib/tlIO/STB.h:59]: SGI -> STB

File: lib/tlIO/STBRead.cpp [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+                    out.time = time;
+                    out.image = imaging::Image::create(_info);
+
+                    const int channels = static_cast<int>(
+                        imaging::getChannelCount(_info.pixelType));
+                    const size_t bytes = imaging::getBitDepth(_info.pixelType) / 8;
+                    
+                    stbi_set_flip_vertically_on_load(1);
+
+                    int x, y, n;
```
Comment (darbyjohnston) [inline lib/tlIO/STBRead.cpp:70]: It would be good to initialize these variables to zero.

File: lib/tlIO/STBRead.cpp [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        namespace
+        {
+            
+            class File
+            {
+            public:
+                File(const std::string& fileName, const file::MemoryRead* memory)
+                {
+                    int w = 0, h = 0, n = 0, bits = 8;
+                    int res = stbi_info(fileName.c_str(), &w, &h, &n);
```
Comment (darbyjohnston) [inline lib/tlIO/STBRead.cpp:29]: Does STB have the ability to read from memory and files?

**Review by darbyjohnston — CHANGES_REQUESTED** [review]
Looks like an interesting addition, though to be honest I was hoping to leave some of the older formats like TGA behind (DJV has TGA support which I did not port over to tlRender). On the other hand it's not a huge amount of code which is good.

My main concern would be if STB handles reading images from memory? That's necessary for the .otioz support which is essential. Also tests would need to be added similar to the other image formats in "tests\tlIOTest".

File: lib/tlIO/CMakeLists.txt [hunk 5]
```diff
@@ -1,4 +1,6 @@
 set(HEADERS
+    stb_image.h
```
[the remark this answers is not in the document]
Comment (ggarra13) [reply lib/tlIO/CMakeLists.txt]: I moved the .h files to deps/stb.

File: lib/tlIO/STBRead.cpp (hunk 4 above)
Comment (ggarra13) [reply lib/tlIO/STBRead.cpp]: Yes.  I added new code to handle it now.

File: lib/tlIO/STBWrite.cpp [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+                    stbi_flip_vertically_on_write(1);
+    
+                    file::Path path(fileName);
+                    std::string ext = path.getExtension();
+                    if (string::compareNoCase(ext, ".tga"))
+                        stbi_write_tga(fileName.c_str(), info.size.w,
+                                       info.size.h, comp, image->getData());
+                    else if (string::compareNoCase(ext, ".bmp"))
+                        stbi_write_bmp(fileName.c_str(), info.size.w,
```
[the remark this answers is not in the document]
Comment (ggarra13) [reply lib/tlIO/STBWrite.cpp:43]: No.  There's currently no library that handles writing PSD files, unfortunately.

File: CMakeLists.txt (hunk 1 above)
Comment (ggarra13) [reply CMakeLists.txt]: Fixed.

**Comment by ggarra13** [comment]
> My main concern would be if STB handles reading images from memory? That's necessary for the .otioz support which is essential. Also tests would need to be added similar to the other image formats in "tests\tlIOTest".

It does handle images from memory.  I added code for it now, but I didn't know how to test it.  I also added a STBTest.cpp/.h based on the SGITest but also didn't know how to test it.  Can you test them and see if they are okay?

**Comment by darbyjohnston** [comment]
You can run the tests like:
* make test (when using the CMake Makefile generator)
* run the test binary directly: tests/tltest/tltest

To test reading from memory you can use an .otioz file. However the "otioconvert" utility doesn't support image sequences yet, but I have a PR to fix that:
https://github.com/AcademySoftwareFoundation/OpenTimelineIO/pull/1627

File: lib/tlIO/STBWrite.cpp [hunk 7]
```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2021-2023 Darby Johnston
+// All rights reserved.
+
+#define STBIW_WINDOWS_UTF8
```
Comment (darbyjohnston) [inline lib/tlIO/STBWrite.cpp:5]: Just checking about these defines, they are different for reading and writing? The write doesn't need "STBI_NO_JPEG" and read doesn't need "STBIW_WINDOWS_UTF8"?

File: lib/tlIO/STBRead.cpp [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+    namespace stb
+    {
+        namespace
+        {
+            
+            class File
+            {
+            public:
+                File(const std::string& fileName, const file::MemoryRead* memory)
+                    {
```
Comment (darbyjohnston) [inline lib/tlIO/STBRead.cpp:27]: It looks like this block of code got extra indentation, the opening and closing brackets are not aligned.

File: lib/tlIO/STBWrite.cpp (hunk 7 above)
Comment (ggarra13) [reply lib/tlIO/STBWrite.cpp:5]: The NO_JPEG, etc. don't do anything in the stb_image_write.h file.  However you are right that I missed the UTF8 define for stb_image.h.  I fixed it (Good catch!).

File: lib/tlIO/STBRead.cpp (hunk 8 above)
Comment (ggarra13) [reply lib/tlIO/STBRead.cpp]: Fixed.

**Comment by darbyjohnston** [comment]
Thanks!
